addons/sale/controllers/variant.py

```diff
@@ -10,7 +10,9 @@ class VariantController(http.Controller):
     def get_combination_info(self, product_template_id, product_id, combination, add_qty, pricelist_id, **kw):
         combination = request.env['product.template.attribute.value'].browse(combination)
         pricelist = self._get_pricelist(pricelist_id)
-        ProductTemplate = request.env['product.template']
+        cids = request.httprequest.cookies.get('cids', str(request.env.user.company_id.id))
+        allowed_company_ids = [int(cid) for cid in cids.split(',')]
+        ProductTemplate = request.env['product.template'].with_context(allowed_company_ids=allowed_company_ids)
```

Comment:
I suppose It's a way to do it :D.
Did you try fixing the problem in the js rpc call directly ?